config/routes.rb

```diff
@@ -80,6 +80,7 @@
     resources :extra_mobile_data_requests, only: %i[index show edit update], path: '/extra-mobile-data-requests' do
       put 'bulk-update', to: 'extra_mobile_data_requests#bulk_update', on: :collection
       get 'report-a-problem', to: 'extra_mobile_data_requests#report_problem', as: :report_problem
+      put 'find-requests', to: 'extra_mobile_data_requests#find_requests', on: :collection
```

Comment:
I'm a little uneasy about using non-safe HTTP verbs for safe actions - IMHO it's almost the same as using safe verbs for unsafe actions - although I guess there could be too may numbers to fit into a URL param, but in that case it might be more conventional to fall back to POST. Was there a reason why you chose PUT?

Reply:
Agree, it should be a POST